File: app/src/main/java/com/imcys/bilibilias/home/ui/model/UserCreateCollectionBean.java
package com.imcys.bilibilias.home.ui.model;

import java.io.Serializable;
import java.util.List;

/**
 * @author:imcys
 * @create: 2022-11-16 20:19
 * @Description: 用户创建收藏夹数据
 */


public class UserCreateCollectionBean implements Serializable {

    /**
     * code : 0
     * message : 0
     * ttl : 1
     * data : {"count":6,"list":[{"id":196026807,"fid":1960268,"mid":351201307,"attr":0,"title":"默认收藏夹","fav_state":0,"media_count":452},{"id":1429039607,"fid":14290396,"mid":351201307,"attr":2,"title":"物理学习","fav_state":0,"media_count":1},{"id":1426414307,"fid":14264143,"mid":351201307,"attr":2,"title":"设计学习","fav_state":0,"media_count":1},{"id":1292535107,"fid":12925351,"mid":351201307,"attr":54,"title":"致我童年","fav_state":0,"media_count":8},{"id":1274129407,"fid":12741294,"mid":351201307,"attr":2,"title":"生物学习","fav_state":0,"media_count":3},{"id":1224727107,"fid":12247271,"mid":351201307,"attr":2,"title":"思想政治","fav_state":0,"media_count":11}],"season":null}
     */

    private int code;
    private String message;
    private int ttl;
    private DataBean data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getTtl() {
        return ttl;
    }

    public void setTtl(int ttl) {
        this.ttl = ttl;
    }

    public DataBean getData() {
        return data;
    }

    public void setData(DataBean data) {
        this.data = data;
    }

    public static class DataBean implements Serializable {
        /**
         * count : 6
         * list : [{"id":196026807,"fid":1960268,"mid":351201307,"attr":0,"title":"默认收藏夹","fav_state":0,"media_count":452},{"id":1429039607,"fid":14290396,"mid":351201307,"attr":2,"title":"物理学习","fav_state":0,"media_count":1},{"id":1426414307,"fid":14264143,"mid":351201307,"attr":2,"title":"设计学习","fav_state":0,"media_count":1},{"id":1292535107,"fid":12925351,"mid":351201307,"attr":54,"title":"致我童年","fav_state":0,"media_count":8},{"id":1274129407,"fid":12741294,"mid":351201307,"attr":2,"title":"生物学习","fav_state":0,"media_count":3},{"id":1224727107,"fid":12247271,"mid":351201307,"attr":2,"title":"思想政治","fav_state":0,"media_count":11}]
         * season : null
         */

        private int count;
        private Object season;
        private List<ListBean> list;

        public int getCount() {
            return count;
        }

        public void setCount(int count) {
            this.count = count;
        }

        public Object getSeason() {
            return season;
        }

        public void setSeason(Object season) {
            this.season = season;
        }

        public List<ListBean> getList() {
            return list;
        }

        public void setList(List<ListBean> list) {
            this.list = list;
        }

        public static class ListBean implements Serializable {
            /**
             * id : 196026807
             * fid : 1960268
             * mid : 351201307
             * attr : 0
             * title : 默认收藏夹
             * fav_state : 0
             * media_count : 452
             */

            private long id;
            private long fid;
            private long mid;
            private int attr;
            private String title;
            private int fav_state;
            private int media_count;
            private int selected = 0;

            public int getSelected() {
                return selected;
            }

            public void setSelected(int selected) {
                this.selected = selected;
            }

            public long getId() {
                return id;
            }

            public void setId(long id) {
                this.id = id;
            }

            public long getFid() {
                return fid;
            }

            public void setFid(long fid) {
                this.fid = fid;
            }

            public long getMid() {
                return mid;
            }

            public void setMid(long mid) {
                this.mid = mid;
            }

            public int getAttr() {
                return attr;
            }

            public void setAttr(int attr) {
                this.attr = attr;
            }

            public String getTitle() {
                return title;
            }

            public void setTitle(String title) {
                this.title = title;
            }

            public int getFav_state() {
                return fav_state;
            }

            public void setFav_state(int fav_state) {
                this.fav_state = fav_state;
            }

            public int getMedia_count() {
                return media_count;
            }

            public void setMedia_count(int media_count) {
                this.media_count = media_count;
            }
        }
    }
}

File: app/src/main/java/com/imcys/bilibilias/home/ui/model/UserCreateCollectionBean.kt
package com.imcys.bilibilias.home.ui.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * ![获取指定用户创建的所有收藏夹信息](https://github.com/SocialSisterYi/bilibili-API-collect/blob/master/docs/fav/info.md#%E8%8E%B7%E5%8F%96%E6%8C%87%E5%AE%9A%E7%94%A8%E6%88%B7%E5%88%9B%E5%BB%BA%E7%9A%84%E6%89%80%E6%9C%89%E6%94%B6%E8%97%8F%E5%A4%B9%E4%BF%A1%E6%81%AF)
 * ```
 * {
 *     "code": 0,
 *     "message": "0",
 *     "ttl": 1,
 *     "data": {
 *         "count": 2,
 *         "list": [
 *             {
 *                 "id": 939227072,
 *                 "fid": 9392270,
 *                 "mid": 509372,
 *                 "attr": 54,
 *                 "title": "学习",
 *                 "fav_state": 0,
 *                 "media_count": 22
 *             },
 *             {
 *                 "id": 75020272,
 *                 "fid": 750202,
 *                 "mid": 509372,
 *                 "attr": 22,
 *                 "title": "MAD/AMV",
 *                 "fav_state": 0,
 *                 "media_count": 16
 *             }
 *         ],
 *         "season": null
 *     }
 * }
 * ```
 */
@Serializable
data class UserCreateCollectionBean(
    @SerialName("count")
    val count: Int = 0, // 2
    @SerialName("list")
    val list: List<Collection> = listOf(),
) {
    @Serializable
    data class Collection(
        @SerialName("attr")
        val attr: Int = 0, // 54
        @SerialName("fav_state")
        val favState: Int = 0, // 0
        @SerialName("fid")
        val fid: Int = 0, // 9392270
        @SerialName("id")
        val id: Int = 0, // 939227072
        @SerialName("media_count")
        val mediaCount: Int = 0, // 22
        @SerialName("mid")
        val mid: Int = 0, // 509372
        @SerialName("title")
        val title: String = "" // 学习
    )
}
